src/rules/newlineRule.ts

```diff
@@ -0,0 +1,150 @@
+/**
+* @license
+* Copyright 2017 Palantir Technologies, Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+ 
+import { getPreviousStatement, isBlock, isClassDeclaration, isFunctionWithBody, getChildOfKind } from "tsutils";
+import * as ts from "typescript";
+import * as Lint from "../index";
+ 
+enum OptionState {
+    always = "always",
+    ignore = "ignore",
+    never = "never",
+}
+ 
+const ALWAYS_IGNORE_OR_NEVER = {
+    enum: OptionState,
```

Comment:
this is only used in the metadata? I don't think this works with string enums. It actually expects a string array here.